Build failure: pkgsCross.aarch64-multiplatform-musl.perl

### Nixpkgs version

- Stable (25.05)

### Steps to reproduce

`nix-build '<nixpkgs>' -A pkgsCross.aarch64-multiplatform-musl.perl`

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
error: build of '/nix/store/jsskhsm5bqf60ms8xnqp0mm00prvi86d-perl-aarch64-unknown-linux-musl-5.40.0.drv' on 'ssh-ng://nixdistributed@mainserv' failed: builder for '/nix/store/jsskhsm5bqf60ms8xnqp0mm00prvi86d-perl-aarch64-unknown-linux-musl-5.40.0.drv' failed with exit code 2;
       last 25 log lines:
       >                  from op.c:163:
       > /nix/store/c9z8w137q6xm10d8kcg105ky8rwixk1r-aarch64-unknown-linux-musl-gcc-14.2.1.20250322/aarch64-unknown-linux-musl/sys-include/sys/cdefs.h:1:2: warning: #warning usage of non-standard #include <sys/cdefs.h> is deprecated [-Wcpp]
       >     1 | #warning usage of non-standard #include <sys/cdefs.h> is deprecated
       >       |  ^~~~~~~
       > op.c: In function 'Perl_newATTRSUB_x':
       > perl.h:2399:25: error: implicit declaration of function 'memrchr'; did you mean 'memchr'? [-Wimplicit-function-declaration]
       >  2399 | #    define my_memrchr  memrchr
       >       |                         ^~~~~~~
       > op.c:10752:34: note: in expansion of macro 'my_memrchr'
       > 10752 |         const char *s = (char *) my_memrchr(name, ':', namlen);
       >       |                                  ^~~~~~~~~~
       > op.c:10752:25: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
       > 10752 |         const char *s = (char *) my_memrchr(name, ':', namlen);
       >       |                         ^
       > In file included from perl.h:7874:
       > sv_inline.h: At top level:
       > sv_inline.h:566:1: warning: 'always_inline' function might not be inlinable unless also declared 'inline' [-Wattributes]
       >   566 | Perl_SvPVXtrue(pTHX_ SV *sv)
       >       | ^~~~~~~~~~~~~~
       > sv_inline.h:536:1: warning: 'always_inline' function might not be inlinable unless also declared 'inline' [-Wattributes]
       >   536 | Perl_newSV_type_mortal(pTHX_ const svtype type)
       >       | ^~~~~~~~~~~~~~~~~~~~~~
       > make[1]: *** [Makefile:157: op.o] Error 1
       > make[1]: Leaving directory '/build/perl-5.40.0'
       > make: *** [Makefile:87: all] Error 2
       For full logs, run:
         nix log /nix/store/jsskhsm5bqf60ms8xnqp0mm00prvi86d-perl-aarch64-unknown-linux-musl-5.40.0.drv
error: builder for '/nix/store/jsskhsm5bqf60ms8xnqp0mm00prvi86d-perl-aarch64-unknown-linux-musl-5.40.0.drv' failed with exit code 1
```

### Additional context

I believe this is the same bug as https://bugs.gentoo.org/932385

According to the Gentoo bug report, the fix should be to add `-D_GNU_SOURCE` to the build flags, but I don't know how or where exactly...

### System metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.39, NixOS, 25.05 (Warbler), 25.05.git.0c59af877b4b`
```

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/interpreters/perl/interpreter.nix
+++ b/pkgs/development/interpreters/perl/interpreter.nix
@@ -125,6 +125,8 @@ stdenv.mkDerivation (
             "-Dlibpth=\"\""
             "-Dglibpth=\"\""
             "-Ddefault_inc_excludes_dot"
+            # https://github.com/arsv/perl-cross/issues/158
+            "-Dd_gnulibc=define"
           ]
         else
           (

